q2_mystery_stew/plugin_setup.py

```diff
@@ -129,11 +127,34 @@ def __call__(self, use):
             if argument is None:
                 inputs[name] = transformed_inputs[name] = None
             elif is_semantic_type(template.qiime_type):
-                inputs[name] = use.init_data(argument.__name__, argument)
-                artifact = argument()
-                view = artifact.view(template.view_type)
-                view.__hide_from_garbage_collector = artifact
-                transformed_inputs[name] = view
+                if type(argument) == list:
+                    input_temp = []
+                    transformed_inputs[name] = []
+                    for arg in argument:
+                        input_temp.append(use.init_data(arg.__name__, arg))
+                        artifact = arg()
+                        view = artifact.view(template.view_type)
+                        view.__hide_from_garbage_collector = artifact
+                        transformed_inputs[name].append(view)
+                    inputs[name] = use.init_data_collection(name, list,
+                                                            *input_temp)
+                elif type(argument) == set:
+                    input_temp = []
+                    transformed_inputs[name] = set()
+                    for arg in argument:
+                        input_temp.append(use.init_data(arg.__name__, arg))
+                        artifact = arg()
+                        view = artifact.view(template.view_type)
+                        view.__hide_from_garbage_collector = artifact
+                        transformed_inputs[name].add(view)
+                    inputs[name] = use.init_data_collection(name, set,
+                                                            *input_temp)
+                else:
+                    inputs[name] = use.init_data(argument.__name__, argument)
+                    artifact = argument()
+                    view = artifact.view(template.view_type)
+                    view.__hide_from_garbage_collector = artifact
+                    transformed_inputs[name] = view
```

Comment:
It may be reasonable to combine this branch with the inner loop of the `for arg in argument` in the above two branches (as they are just initializing the single case in a loop), but these situations often don't end up being much simpler when you refactor them, so if it doesn't look nice, don't worry about it.